tests/v2/validation/charts/webhook_test.go

```diff
@@ -0,0 +1,139 @@
+	w.session = testSession
+
+	client, err := rancher.NewClient("", testSession)
+	require.NoError(w.T(), err)
+
+	w.client = client
+
+	// Get clusterName from config yaml
+	w.clusterList = client.RancherConfig.ClusterName
+	w.chartVersion, err = client.Catalog.GetLatestChartVersion(charts.RancherWebhookName)
+	require.NoError(w.T(), err)
+
+}
+
+func (w *WebhookTestSuite) ValidateEscalationCheck(client *rancher.Client) {
+
+	getAdminRole, err := client.Management.GlobalRole.ByID(admin)
+	require.NoError(w.T(), err)
+	updatedAdminRole := *getAdminRole
+	updatedAdminRole.NewUserDefault = true
+
+	_, err = client.Management.GlobalRole.Update(getAdminRole, updatedAdminRole)
+	require.Error(w.T(), err)
+	errMessage := "admission webhook \"rancher.cattle.io.globalroles.management.cattle.io\" denied the request"
+
+	assert.Contains(w.T(), err.Error(), errMessage)
+}
+
+func (w *WebhookTestSuite) ValidatePodLogs(podName, clusterID string) {
+
+	podLogs, err := kubeconfig.GetPodLogs(w.client, clusterID, podName, charts.RancherWebhookNamespace)
+	require.NoError(w.T(), err)
+	delimiter := "\n"
+	segments := strings.Split(podLogs, delimiter)
+
+	for _, segment := range segments {
+		if strings.Contains(segment, "level=error") {
+			w.Fail("Error logs in webhook", segment)
+		}
+	}
+
+}
+
+func (w *WebhookTestSuite) TestWebhookChart() {
+	subSession := w.session.NewSession()
+	defer subSession.Cleanup()
+
+	clusterID, err := clusters.GetClusterIDByName(w.client, w.clusterList)
+	require.NoError(w.T(), err)
+
+	w.Run("Verify the version of webhook on local and downstream cluster", func() {
+		subSession := w.session.NewSession()
+		defer subSession.Cleanup()
+
+		initialWebhookChart, err := charts.GetChartStatus(w.client, clusterID, charts.RancherWebhookNamespace, charts.RancherWebhookName)
+		require.NoError(w.T(), err)
+		chartVersion := initialWebhookChart.ChartDetails.Spec.Chart.Metadata.Version
+		require.NoError(w.T(), err)
+		assert.Equal(w.T(), w.chartVersion, chartVersion)
+	})
+
+	w.Run("Verify webhook pod logs", func() {
+
+		steveClient := w.client.Steve
+		pods, err := steveClient.SteveType(pods.PodResourceSteveType).NamespacedSteveClient(charts.RancherWebhookNamespace).List(nil)
+		require.NoError(w.T(), err)
+
+		var podName string
+		for _, pod := range pods.Data {
+			if strings.Contains(pod.Name, charts.RancherWebhookName) {
+				podName = pod.Name
+			}
+		}
+		w.ValidatePodLogs(podName, clusterID)
+
+	})
+
+	w.Run("Verify the count of webhook is greater than zero and list webhooks", func() {
+		webhookList, err := getWebhook(w.client, clusterID, resourceName)
+		require.NoError(w.T(), err)
+
+		assert.True(w.T(), len(webhookList) > 0, "Expected webhooks list to be greater than zero")
+		log.Info("Count of webhook obtained for the cluster: ", w.clusterList, "is ", len(webhookList))
+		listStr := strings.Join(webhookList, ", ")
+		log.WithField("", listStr).Info("List of webhooks obtained for the ", w.clusterList)
```

Comment:
Why do we need the list of webhooks ? (i think its the list of controllers here?)

Reply:
its not the list of controllers but a list of admission handlers. so named it as list of webhooks. I'm prining the list of webhooks because for each rancher version there may or may not add an additional webhook depending on the norman to steve changes. By printing the list here, we can keep track of the pre and post upgrade information. This allows us to identify any new handlers that may have been created in the future.